webpack.config.js

```diff
@@ -0,0 +1,182 @@
+  'NODE_ENV'
+);
+
+const BUILD_ENV = t.strict(
+  {
+    NODE_ENV,
+    BUNDLE_TARGET: t.union([t.literal('firefox'), t.literal('chrome')]),
+    BUNDLE_STATS: BooleanFromString,
+  },
+  'processENV'
+);
+
+const buildENV = pipe(
+  {
+    BUNDLE_TARGET: 'chrome',
+    BUNDLE_STATS: 'false',
+    NODE_ENV: mode,
+    ...process.env
+  },
+  BUILD_ENV.decode,
+  (validation) => {
+    if (validation._tag === 'Left') {
+      console.error(PathReporter.report(validation).join('\n'));
+      console.log('\n');
+      throw new Error('process.env decoding failed.');
+    }
+    return validation.right;
+  }
+);
+
+const pkgJson = require('./package.json');
+const manifestVersion = process.env.MANIFEST_VERSION || pkgJson.version.replace('-beta', '');
+
+const plugins = [
+  new DefinePlugin({
+    'process.env.REACT_APP_BUILD_DATE': JSON.stringify(new Date().toISOString()),
+    'process.env.REACT_APP_VERSION': JSON.stringify(manifestVersion),
+    'process.env.NODE_ENV': JSON.stringify(mode),
+  }),
+
+  new DotenvPlugin({
+    path: dotEnvConfigPath,
+  }),
+
+  new CopyWebpackPlugin({
+    patterns: [{
+      from: 'public',
+      filter: (file) => {
+        const { base } = path.parse(file);
+
+        if (base === 'manifest.json') {
+          return false;
+        }
+
+        return true;
+      }
+    }, {
+      from: 'public/manifest.json',
+      transform: (content) => {
+        const manifest = JSON.parse(content);
+
+        if (buildENV.BUNDLE_TARGET === 'chrome') {
+          manifest.cross_origin_embedder_policy = {
+            value: 'require-corp',
+          };
+
+          manifest.cross_origin_opener_policy = {
+            value: 'same-origin',
+          };
+        }
+
+        manifest.version = manifestVersion;
+
+        return JSON.stringify(manifest, null, 2);
+      }
+    }],
+  }),
+];
+
+if (mode === 'production') {
+  plugins.push(new FileManagerPlugin({
+    events: {
+      onEnd: {
+        archive: [{
+          source: './build',
+          destination: './build/extension.zip',
+        }],
+      },
+    },
+  }));
+} else {
+  if (buildENV.BUNDLE_STATS) {
+    plugins.push(new BundleAnalyzerPlugin({
+      generateStatsFile: true,
+      analyzerMode: 'json',
+    }));
+  }
+}
+
+module.exports = {
+  mode,
+
+  entry: {
+    main: path.resolve(__dirname, 'src/index.tsx'),
+    app: path.resolve(__dirname, 'src/app.tsx'),
+    popup: path.resolve(__dirname, 'src/popup.tsx'),
+    background: path.resolve(__dirname, 'src/background/index.ts'),
+  },
+
+  output: {
+    path: path.resolve(__dirname, 'build'),
+    filename: '[name].js',
+  },
+
+  module: {
+    rules: [{
+      test: /\.tsx?$/,
+      use: [{
+        loader: 'ts-loader',
+        options: {
+          compilerOptions: {
+            noEmit: false,
+            sourceMap: true,
+          },
+          transpileOnly: true,
+        },
+      }],
+    }, {
+      test: /\.(ttf|svg)$/,
```

Comment:
no `.png` nor `.jpg` here?

Reply:
we don't need it because we don't load any png or jpg as data URLs